codechecker_common/guidelines.py

```diff
@@ -0,0 +1,77 @@
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+import os
+from typing import Any, DefaultDict, Dict, Iterable, List
+from collections import defaultdict
+
+import yaml
+
+
+class Guidelines:
```

Comment:
There is a function that already provides all guidelines and their rules. 
```
 def all_guideline_rules(self) -> DefaultDict[str, List[Dict[str, str]]]:
        return self.__all_rules
```

Do you think it is enough or it would be better to have a dedicated guideline name function?